control-plane/agents/core/src/volume/specs.rs

```diff
@@ -606,49 +585,90 @@ impl ResourceSpecsLocked {
-            if spec.updating {
-                return Err(SvcError::Conflict {});
-            } else if !spec.state.created() {
-                return Err(SvcError::NexusNotFound {
-                    nexus_id: request.nexus.to_string(),
-                });
-            } else if !spec.children.contains(&request.uri) {
-                return Err(SvcError::ChildNotFound {
-                    nexus: request.nexus.to_string(),
-                    child: request.uri.to_string(),
-                });
+            let spec_clone = {
+                let status = registry.get_nexus(&request.nexus).await?;
+                let mut spec = nexus_spec.lock().await;
+                if spec.pending_op() {
+                    return Err(SvcError::StoreSave {
+                        kind: ResourceKind::Nexus,
+                        id: request.nexus.to_string(),
+                    });
+                } else if spec.updating {
+                    return Err(SvcError::Conflict {});
+                } else if !spec.state.created() {
+                    return Err(SvcError::NexusNotFound {
+                        nexus_id: request.nexus.to_string(),
+                    });
+                } else if !spec.children.contains(&request.uri)
+                    && !status.children.iter().any(|c| c.uri == request.uri)
+                {
+                    return Err(SvcError::ChildNotFound {
+                        nexus: request.nexus.to_string(),
+                        child: request.uri.to_string(),
+                    });
+                }
+
+                spec.updating = true;
+                spec.start_op(NexusOperation::RemoveChild(request.uri.clone()));
+                spec.clone()
+            };
+
+            if let Err(error) = registry.store_obj(&spec_clone).await {
+                let mut spec = nexus_spec.lock().await;
+                spec.updating = false;
+                spec.clear_op();
+                return Err(error);
             }
 
-            spec.updating = true;
-            let mut spec_clone = spec.clone();
-            drop(spec);
+            let result = node.remove_child(request).await;
+            Self::nexus_complete_op(registry, result, nexus_spec, spec_clone).await
+        } else {
+            node.remove_child(request).await
+        }
+    }
 
-            match node.remove_child(request).await {
-                Ok(_) => {
-                    spec_clone.children.retain(|c| c != &request.uri);
-                    let result = {
-                        let mut store = registry.store.lock().await;
-                        store.put_obj(&spec_clone).await
-                    };
-                    if let Err(error) = result {
-                        let mut spec = nexus_spec.lock().await;
-                        spec.updating = false;
-                        return Err(error.into());
+    /// Completes a nexus update operation by trying to update the spec with the persistent store.
```

Comment:
I think it is meant to say "...update the spec **in** the persistent store."

Reply:
Done